randovania/game_connection/connector/am2r_remote_connector.py

```diff
@@ -146,32 +150,37 @@ def new_inventory_received(self, new_inventory: str) -> None:
         Regarding ammo and progressives: for some reason, we need here the name of the ammo, and not the name of the
         expansions. I.e. for `Missile Expansions`, we need `Missiles`. So there's a lookup dict here to adjust for them.
 
-        :param new_inventory: A string from the game with the format `items:XA|XB,YA|YB,ZA|ZB...`.
+        :param new_inventory_response: A string from the game with the format `items:XA|XB,YA|YB,ZA|ZB...`.
         """
+        print(f"TEST: {new_inventory_response}")
         if self.current_region is None:
             return
 
         inventory_dict = defaultdict(int)
         start_of_inventory = "items:"
 
-        if not new_inventory.startswith(start_of_inventory):
-            self.logger.warning("Unknown response: %s", new_inventory)
+        if not new_inventory_response.startswith(start_of_inventory):
+            self.logger.warning("Unknown response: %s", new_inventory_response)
             return
 
-        for position in new_inventory[len(start_of_inventory) :].split(","):
+        inventory_list = new_inventory_response[len(start_of_inventory) :].split(",")
+        if len(inventory_list[-1]) == 0 or inventory_list[-1].isspace():
```

Comment:
`not inventory_list[-1]`?

Reply:
string truthyness is `true` for a non-empty whitespace string.
```
>>> test = "             "
>>> if test:
...  print("true")
... 
true
```

Reply:
I could make it as `not inventory_list[-1] or inventory_list[-1].isspace()` but I don't think that's clearer.
Alternatively, i'll just write a small function for it, so that it reads as `if not is_empty_or_space(inventory_list[-1])`

Reply:
Ah I misread what you wanted, yeah if you want to catch any number of trailing spaces then yeah strip it

Reply:
oh yeah that'd do it